judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/submission/bundle/ContestItemSubmissionResource.java

```diff
@@ -92,19 +96,30 @@ public ContestItemSubmissionResource(
     public ContestItemSubmissionsResponse getSubmissions(
             AuthHeader authHeader,
             String contestJid,
-            Optional<String> userJid,
-            Optional<String> problemJid,
+            Optional<String> username,
+            Optional<String> problemAlias,
             Optional<Integer> page) {
 
         String actorJid = actorChecker.check(authHeader);
         Contest contest = checkFound(contestStore.getContestByJid(contestJid));
         checkAllowed(submissionRoleChecker.canViewOwn(actorJid, contest));
 
         boolean canSupervise = submissionRoleChecker.canSupervise(actorJid, contest);
-        Optional<String> actualUserJid = canSupervise ? userJid : Optional.of(actorJid);
+        Optional<String> filterUserJid;
+        if (canSupervise) {
+            filterUserJid = username.map(u -> userSearchService.translateUsernamesToJids(ImmutableSet.of(u)).get(u));
```

Comment:
If `username` / `problemAlias` is invalid, this endpoint should not return 404, but an empty data instead.

We can for example use `Optional.of("")` for invalid `username` / problemAlias`

Reply:
OK, fixed